deployment/ccip/changeset/cs_active_candidate.go

```diff
@@ -17,37 +17,68 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
 )
 
+var (
+	_ deployment.ChangeSet[PromoteAllCandidatesChangesetConfig]  = PromoteAllCandidatesChangeset
+	_ deployment.ChangeSet[AddDonAndSetCandidateChangesetConfig] = SetCandidatePluginChangeset
+)
+
+type PromoteAllCandidatesChangesetConfig struct {
+	HomeChainSelector uint64
+	NewChainSelector  uint64
+	Nodes             deployment.Nodes
```

Comment:
It'll be a lot easier to invoke in cld if this just takes nodeIDs instead and loads the relevant info via the OffchainClient in the changeset

Reply:
Makes sense, by node IDs you mean P2P IDs?